zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -95,12 +95,18 @@ fn main() -> Result<()> {
         .map(BlockHeight)
         .expect("zcashd has some mature blocks: wait for zcashd to sync more blocks");
 
+    let mut starting_height = BlockHeight(args::Args::from_args().last_checkpoint);
```

Comment:
Issue: We're missing a range check here.

Suggestion: Let's make the genesis special case part of the type (2/3)
Step 2:
  * Use a mapping to transform `Some(h)` to `Some(BlockHeight(h))`
  * Do the range check on `Some(BlockHeight(h))`

```suggestion
    let mut starting_height = args::Args::from_args().last_checkpoint.map(BlockHeight);
    if starting_height.is_some() {
        // Since we're about to add 1, height needs to be strictly less than the maximum
        assert!(starting_height.unwrap() < BlockHeight::MAX);
    }
```